src/Symfony/Component/Config/Definition/Builder/NodeDefinition.php

```diff
@@ -178,6 +178,11 @@ public function getNode(bool $forceRootNode = false): NodeInterface
      */
     public function defaultValue(mixed $value): static
     {
+        if ($this->required) {
+            // throw new InvalidDefinitionException(sprintf('You cannot set a default value to the node "%s" as it is required.', $this->name));
+            trigger_deprecation('symfony/config', '7.4', 'Setting a default value to the node "%s" is deprecated when it is required.', $this->name);
```

Comment:
"it" is a bit ambigous in this context.

```suggestion
            trigger_deprecation('symfony/config', '7.4', 'Setting a default value to a required node is deprecated. Remove the default from node "%s" or make it optional.', $this->name);
```